lib/MetaCPAN/Script/Mapping.pm

```diff
@@ -132,13 +141,18 @@ sub run {
     my $self = shift;
 
     if ( $self->await ) {
-        $self->delete_index if $self->arg_delete_index;
```

Comment:
why the change of order? maybe we shouldn't allow a mix of these flags and have an if-else construct here to prevent that.

Reply:
I have implemented the ` if-else ` check for all **editing operations** as requested.
I hope I understood well what you had in mind.
Please note that I also introduced **several tests** to check the editing operations.

Reply:
thanks, I think that's better. if we ever need to mix flags we can change it accordingly, but that's a safer starting point.